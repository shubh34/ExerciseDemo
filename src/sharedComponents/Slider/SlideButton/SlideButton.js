import React from 'react';
import './SlideButton.scss';

const SlideButton = ({ onClick, type }) => {
	const arrowIcon = type === 'prev' ? 'icon__left-nav' : 'icon__right-nav';
	return (
		<button type="button" className={`slide-button slide-button--${type}`} onClick={onClick}>
			<span className={arrowIcon} />
		</button>
	);
};
export default SlideButton;
