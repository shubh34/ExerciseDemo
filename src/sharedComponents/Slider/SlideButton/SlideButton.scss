.slide-button {
    position: absolute;
    top: 0;
    bottom: 0;
    width: 48px;
    background: rgba(0, 0, 0, 0.10);
    border: 0;
    outline: 0;
    padding: 0;
    z-index: 4;
    cursor: pointer;
    font-size: 40px;
  
    span {
      color: #fff;
      display: block;
      margin: 0 auto;
      text-shadow: 0 0 3px rgba(0,0,0,.8);
    }
  
    &--next {
      right: 0;
  
      span ::before {
        content: '\e637'
      }
    }
  
    &--prev {
      left: 0;
    }
  }
  