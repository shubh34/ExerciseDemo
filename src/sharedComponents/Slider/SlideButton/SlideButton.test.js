
import React from 'react';
import { shallow } from 'enzyme';
import SlideButton from './SlideButton';

describe('SlideButton', () => {
	it('renders without crashing', () => {
		const wrapper = shallow(<SlideButton type="next" onClick={jest.fn()} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should call onClick on button click', () => {
		const onClick = jest.fn();
		const wrapper = shallow(<SlideButton type="prev" onClick={onClick} />);
		wrapper.find('button').simulate('click');
		expect(onClick).toBeCalled();
	});
});
