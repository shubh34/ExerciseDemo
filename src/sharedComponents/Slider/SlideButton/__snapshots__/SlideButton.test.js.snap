// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideButton renders without crashing 1`] = `
<button
  className="slide-button slide-button--next"
  name="slide-next"
  onClick={[MockFunction]}
  type="button"
>
  <span
    className="icon__right-nav"
  />
</button>
`;
